waypoint_generator/src/waypoint_generator/waypoint_generator_node.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <waypoint_generator/waypoint_generator_node.hpp>
+#include <waypoint_generator/waypoint_generator.hpp>
+
+namespace waypoint_generator
+{
+    void WaypointGeneratorNode::initialize()
+    {
+        ros::init(argc, argv, node_name);
+        _nh.reset(new ros::CARMANodeHandle());
+        _pnh.reset(new ros::CARMANodeHandle("~"));
+        _wm = _wml.getWorldModel();
+        ROS_DEBUG("Initialized all node handles");
+
+        _pnh->param<double>("curvature_epsilon", _curvature_epsilon, 3.0);
+        _pnh->param<int>("linearity_constraint", _linearity_constraint, 2);
+        _pnh->param<double>("lateral_accel_limit", _lateral_accel_limit, 1.5);
+        _pnh->param<double>("longitudinal_accel_limit", _longitudinal_accel_limit, 1.5);
+        _pnh->param<double>("longitudinal_decel_limit", _longitudinal_decel_limit, 1.5);
```

Comment:
Do you think it makes sense to make those three variable as vehicle specific and stored in carma config repo?

Reply:
Maybe? I hadn't thought about it that way, since it's more like the "ideal" speeds/limits for the roadway rather than the limits of the particular vehicle, but that still might make sense. We want these limits to be well below what the vehicle/controller is capable of.